pkg/asset/agent/manifests/clusterimageset_test.go

```diff
@@ -17,34 +18,44 @@ import (
 )
 
 func TestClusterImageSet_Generate(t *testing.T) {
-
 	cases := []struct {
 		name           string
 		dependencies   []asset.Asset
 		expectedError  string
 		expectedConfig *hivev1.ClusterImageSet
 	}{
 		{
-			name:          "missing-config",
+			name: "missing install config",
+			dependencies: []asset.Asset{
+				&agent.OptionalInstallConfig{},
+				&releaseimage.Image{},
+			},
 			expectedError: "missing configuration or manifest file",
 		},
-		// {
-		// 	name: "default",
-		// 	dependencies: []asset.Asset{
-		// 		&releaseimage.Image{
-		// 			PullSpec: "registry.ci.openshift.org/ocp/release:4.11.0-0.nightly-2022-06-04-014713",
-		// 		},
-		// 	},
-		// 	expectedConfig: &hivev1.ClusterImageSet{
-		// 		ObjectMeta: v1.ObjectMeta{
-		// 			Name:      "openshift-was not built correctly",
-		// 			Namespace: "",
-		// 		},
-		// 		Spec: hivev1.ClusterImageSetSpec{
-		// 			ReleaseImage: "registry.ci.openshift.org/ocp/release:4.11.0-0.nightly-2022-06-04-014713",
-		// 		},
-		// 	},
-		// },
+		{
+			name: "invalid ClusterImageSet configuration",
+			dependencies: []asset.Asset{
+				GetValidOptionalInstallConfig(),
+				&releaseimage.Image{},
+			},
+			expectedError: "invalid ClusterImageSet configuration: Spec.ReleaseImage: Forbidden: value must be equal to registry.ci.openshift.org/origin/release:4.11",
+		},
+		{
+			name: "valid configuration",
+			dependencies: []asset.Asset{
+				GetValidOptionalInstallConfig(),
+				GetValidReleaseimage(),
```

Comment:
```suggestion
				&releaseimage.Image{
					PullSpec: "registry.ci.openshift.org/origin/release:4.11",
				},
```
We should avoid mixing test and production code also, the test will break when `defaultReleaseImageOriginal` will get automatically updated in the next release